pyccel/ast/literals.py

```diff
@@ -474,6 +474,8 @@ def convert_to_literal(value, dtype = None):
     if isinstance(primitive_type, PrimitiveIntegerType):
         if value >= 0:
             literal_val = LiteralInteger(value, dtype)
+        elif value == -9223372036854775808:
+            literal_val =  PyccelMinus(PyccelUnarySub(LiteralInteger(-value-1, dtype)), LiteralInteger(1))
```

Comment:
:vomiting_face:

Reply:
Why is this necessary? Don't integers in Python adapt their precision to store any number?

Reply:
This removes the following warning:
```python
if __name__ == '__main__':
    x = -9223372036854775808
```
```bash
/home/jalal/Documents/pyccel_work/pyccel/codegen/compiling/compilers.py:505: UserWarning: /home/jalal/Documents/pyccel/__pyccel__/prog_boo.c: In function ‘main’:
/home/jalal/Documents/pyccel/__pyccel__/prog_boo.c:9:1: warning: integer constant is so large that it is unsigned
    9 |     x = -INT64_C(9223372036854775808);
      | ^   ~~~~~~~~~~~~~~~~

  warnings.warn(UserWarning(err))
  ```
  `9223372036854775808` should not be represented without a `-` sign

Reply:
Ah interesting. I would expect this to be printed as:
```c
x = INT64_C(-9223372036854775808);
```
I will create an issue

Reply:
Actually moving `INT64_C` doesn't help:
```
/home/emily/Code/test/tmp/__pyccel__/prog_tmp_val.c:7:1: warning: integer constant is so large that it is unsigned
    7 |     x = INT64_C(-9223372036854775808);
      | ^   ~~~~~~~~~~~~~~~~
```
Fortran also raises a similar error:
```
/home/emily/Code/test/tmp/__pyccel__/prog_tmp_val.f90:11:30:

   11 |   x = -9223372036854775808_i64
      |                              1
Error: Integer too big for its kind at (1). This check can be disabled with the option ‘-fno-range-check’
```

Reply:
I guess it would make more sense to use : https://en.cppreference.com/w/c/types/limits
and `Huge` in fortran (see https://github.com/pyccel/pyccel/blob/b27a7a8eb3df2a65a1d0bb4004edbf3419884ac2/pyccel/codegen/printing/fcode.py#L2174-L2178)

Related to the comment below where I would rather have this in a separate PR

Reply:
I will update this in a separate issue